From 38670037b194f9a17e779de434384cf3bb239027 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Mon, 2 Mar 2015 20:27:37 +0530
Subject: [PATCH] EM: Release usb mux gpio descriptor for further use.

The usb mux gpio descriptor should be release for further
gpio operation in this gpio line.

Change-Id: Iae237b8c221a22f09c853f8e50186a05cf33fabb
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1117
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/mfd/intel_soc_pmic_dc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mfd/intel_soc_pmic_dc.c b/drivers/mfd/intel_soc_pmic_dc.c
index 7abf130..239f3fb 100644
--- a/drivers/mfd/intel_soc_pmic_dc.c
+++ b/drivers/mfd/intel_soc_pmic_dc.c
@@ -589,6 +589,8 @@ static void dc_xpwr_pwrsrc_pdata(void)
 				"pmic_res", GPIO_USB_MUX_INDEX);
 	if (IS_ERR(pdata.gpio_mux_cntl))
 		pdata.gpio_mux_cntl = NULL;
+	else
+		gpiod_put(pdata.gpio_mux_cntl);
 
 	intel_soc_pmic_set_pdata("dollar_cove_pwrsrc",
 				 (void *)&pdata, sizeof(pdata), 0);
-- 
1.9.1

